lib/tree.rb

```diff
@@ -1,38 +1,63 @@
 class NoApplesError < StandardError; end
 
-class AppleTree
-  attr_#fill_in :height, :age, :apples, :alive
+class Tree
+  attr_accessor :height, :age, :apples, :alive
 
   def initialize
+    @age = 0
+    @height = 0
+    @apples = []
+    @alive = true
   end
 
   def age!
+    @age = @age + 1
+    @height = @height + 10
+    if age > 2
+      add_apples!
+    end
+    if age > 5
+      @alive = false
+    end
   end
 
-  def add_apples
+  def add_apples!
+    i = 0
+    until i > 6 + Random.rand(4)
+      @apples.push(Apple.new('Red', 3 + Random.rand(2))) 
+      i = i + 1
+    end
   end
 
   def any_apples?
+    return !apples.empty?
```

Comment:
The final value in a method will be the return value. Explicit `return` is usually reserved for guard clauses.